torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1939,16 +1945,26 @@ void InitXlaModuleBindings(py::module m) {
         [](const at::Tensor& input, xla::OpSharding sharding) {
           ShardingUtil::XlaMarkSharding(input, sharding);
         });
-  m.def("_mark_manual_sharding", [](const at::Tensor& input,
-                                    xla::OpSharding sharding) {
+  m.def("_mark_manual_sharding",
+        [](const at::Tensor& input, xla::OpSharding sharding) {
+          XLA_CHECK(IsIr(input))
+              << "Marking any data tensors as manual is not supported";
+          ShardingUtil::XlaMarkSharding(input, sharding);
+        });
+  m.def("_spmd_full_to_shard_shape", [](const at::Tensor& input) -> at::Tensor {
```

Comment:
this method is call `xxx_shape` but it actually returns a tensor which is very confusing...

Reply:
I just faithfully copy the XlaOP name... This won't be exposed to the user so I guess it's okay to leave it as it is.

Reply:
yea.. as long as it is just a pybind I am fine with it..